src/main/java/org/zalando/nakadi/controller/EventStreamController.java

```diff
@@ -178,26 +208,21 @@ public StreamingResponseBody streamEvents(
                 consumerCounter = metricRegistry.counter(metricNameFor(eventTypeName, CONSUMERS_COUNT_METRIC_NAME));
                 consumerCounter.inc();
 
-                eventConsumer = topicRepository.createEventConsumer(topic, cursors);
-
-                final Map<String, String> streamCursors = cursors
-                        .stream()
-                        .collect(Collectors.toMap(
-                                Cursor::getPartition,
-                                Cursor::getOffset));
-
-                final EventStreamConfig streamConfig = builder
-                        .withCursors(streamCursors)
-                        .build();
-
                 response.setStatus(HttpStatus.OK.value());
                 response.setContentType("application/x-json-stream");
-                final EventStream eventStream = eventStreamFactory.createEventStream(eventConsumer, outputStream,
-                        streamConfig, blacklistService);
+                final EventConsumer eventConsumer = topicRepository.createEventConsumer(streamConfig.getCursors());
+                eventStream = eventStreamFactory.createEventStream(
+                        outputStream, eventConsumer, streamConfig, blacklistService);
 
                 outputStream.flush(); // Flush status code to client
 
                 eventStream.streamEvents(connectionReady);
+            } catch (final UnparseableCursorException e) {
+                LOG.debug("Incorrect syntax of X-nakadi-cursors header: {}. Respond with BAD_REQUEST.",
+                        e.getCursors(), e);
+                writeProblemResponse(response, outputStream, BAD_REQUEST,
```

Comment:
@rcillo On line 109 it was an exception. Probably I should get message from there.